rtos/source/Thread.cpp

```diff
@@ -113,7 +115,7 @@ osStatus Thread::start(mbed::Callback<void()> task)
     _tid = osThreadNew(Thread::_thunk, this, &_attr);
     if (_tid == nullptr) {
         if (_dynamic_stack) {
-            delete[] _attr.stack_mem;
+            free(_attr.stack_mem);
```

Comment:
No, not technically legal either. If it's allocated with `new[]`, it has to be deleted with `delete[]`.

You have to either put the cast in to name the type of the new:

    delete[] static_cast<uint32_t *>(_attr.stack_mem);

Or actually arrange for the pointer to be a `uint32_t *`. Putting the cast in is probably simplest for now, even if ugly.

I have a bad feeling I might have deleted the cast from there at some point, not realising it was necessary. Might be worth putting a small comment that `delete[]` does not accept `void *`.

Reply:
Thanks. Done.
https://github.com/ARMmbed/mbed-os/compare/64f693c4a0750f03e1fdc30881be224ce73fd591..d25dc43b18a512c294ae84ef07b04ad8a2b10719